### DISCUSSION TITLE
Issue with $wire on Alpine Sort plugin

### DISCUSSION THREAD

**Discussion by johnrudolph** [livewire/livewire#8545] [7. Bugs (archived)]
### Livewire version

v3.3

### Laravel version

v10.10

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

1. [Install Alpine Sort plugin](https://alpinejs.dev/plugins/sort)
2. Create a Livewire component with a public collection and a function you want to call, like below: 

```
<?php

namespace App\Livewire;

use Livewire\Component;
use Illuminate\Support\Collection;

class ListExample extends Component
{
    public Collection $things;

    public function mount()
    {
        $this->things = collect([
            0 => 'thing_1',
            1 => 'thing_2',
            2 => 'thing_3',
        ]);
    }

    public function dieAndDump()
    {
        dd('die and dump');
    }

    public function render()
    {
        return view('livewire.list-example');
    }
}
```

3. In the blade component, create a simple implementation like below: 

```
<div>
    <div x-data="{ handle: (item, position) => { $wire.dieAndDump() } }">
        <ul x-sort="handle">
            @foreach ($things as $item)
                <li x-sort:item>{{ $item }}</li>
            @endforeach
        </ul>
    </div>
</div>
```

4. When you sort the items in your browser, note that your console shows the following:

![Screenshot 2024-06-07 at 4 25 27 PM](https://github.com/livewire/livewire/assets/18694510/2c1aefed-940d-4c72-9dbe-2ac8732a8b05)

5. Then, if you change `$wire` to say `this.$wire` you get a new error in the console: 

![Screenshot 2024-06-07 at 4 27 42 PM](https://github.com/livewire/livewire/assets/18694510/ee3fee03-b8a4-4939-9abb-56bf9f8d6265)

### Repository or Wirebox URL

https://github.com/kathunk/papercuts/pull/13/files

### Additional information

Sorry, I know this is a private repo. All the relevant code is in my steps to reproduce.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by lrljoe** [comment]
You need to tell it what the item is (i.e. have something as the value of sort:item if you want to handle it, otherwise you're trying to pass it null values for "item".

You'll see in the example below, that I'm setting the "item" to be the array index, but you can of course pass in the actual item should you choose.

```
        <div x-data="{ handle: (item, position) => { $wire.dieAndDump(); } }">
            <ul x-sort="handle">
                @foreach ($things as $index => $item)
                    <li x-sort:item="{{ $index }}">{{ $item }}</li>
                @endforeach
            </ul>
        </div>
```

The above works fine on my end using your component as an example.
